### ISSUE TITLE
How to create detached variables?

### ISSUE THREAD

**Issue by mplxrg** [EnzymeAD/Enzyme#311]
In PyTorch, we can detach a tensor from the computation graph using` tensor.detach()`. I wondered if it is possible to achieve similar thing in Enzyme.

**Comment by haianyt** [comment]
Option B is exactly what I was seeking. `__enzyme_inactivefn` seems to be a new feature in v0.0.17 which works pretty well for my use case. One thing to note is that `__attribute__((noinline))` doesn't work for my compiler. The function still gets inlined. I opted to use `__attribute__((optnone))` to prevent the optimization.

**Comment by mplxrg** [comment]
Option B is exactly what I'm seeking. `__enzyme_inactivefn` seems to be a new feature in v0.0.17 which works pretty well for my use case. One thing to note here is that `__attribute((noinline))` doesn't work for my compiler. The function still gets inlined. I opted to use `__attribute((optnone))` to prevent the optimization.

**Comment by wsmoses** [comment]
Circling back on this, I'd be interested in making this an intrinsic directly (e.g. an `__enzyme_inactiveval` or something). Thoughts on name?

**Comment by mplxrg** [comment]
That's would be great! `__enzyme_inactiveval` sounds good to me. Another name I could come up with is `__enzyme_detach()` which is similar to torch.detach().
